### ISSUE TITLE
Grid centers of Nemo grid (NETM) not inside grid corners

### ISSUE THREAD

**Issue by jmrgonza** [uwefladrich/rdy2cpl#21]
While I was plotting the grid definitions for Nemo (NETM grid), I noticed that, for the first column of grid cells, the grid centers are not inside the polygon given by the grid corners. This is for the grid cells with the first coordinate being 0. For example, for cell (0,10), the center coordinates are:
[ 72.5        -76.19991179]
and the corners are:
[ 73.         -76.08013759]
[ 74.         -76.08013759]
[ 74.         -76.31867523]
[ 73.         -76.31867523]
I attach a plot:
![image](https://github.com/uwefladrich/rdy2cpl/assets/5628228/17d7ca86-4be0-447b-b593-35616235b65b)

**Comment by jmrgonza** [comment]
Also, it looks like the grid cells in the first row have similar issues. For example, for cell (10,0) the center coordinates are:
[ 82.5        -78.39350318]
and the corners are:
[[ 83.         -78.29247755]
 [ 82.         -78.29247755]
 [ 82.         -78.29600261]
 [ 83.         -78.29600261]]
In this case, the center is outside the latitude bounds, while in the other it was the longitudes.

I think that it is the coordinates of the corners what is wrong in these two cases.

**Comment by jmrgonza** [comment]
Using this file, the coordinates for the [0,10] grid cell look fine:
Center: [ 73.5        -83.10919952]
Corners: [[ 74.         -83.0489502 ] [ 73.         -83.0489502 ] [ 73.         -83.16892242] [ 74.         -83.16892242]]

but still not for the [10,0] grid cell:
Center: [ 83.5        -84.21070862]
Corners: [[ 84.         -84.16005707] [ 83.         -84.16005707] [ 83.         -84.1618576 ] [ 84.         -84.1618576 ]]

There seems to remain an issue with the longitudes of the corners in the last case (first row of data).

**Comment by jmrgonza** [comment]
I have created a plot with the coordinates of the points in the domain_cfg.nc file. This one is for the eORCA1 domain:
![image](https://github.com/uwefladrich/rdy2cpl/assets/5628228/4bf0b526-6ef8-490e-92ee-cd93027342a9)
Black points are (glamt, gphit), and red points are (glamf, gphif). Black points  should correspond to centers of the Nemo T grid cells. I think that the code in `rdy2cpl/grids/base/orca.py` assumes that the red points correpond to the corner 0 of that grid, with the numbering in:
https://github.com/uwefladrich/rdy2cpl/blob/f60e2618d2d6ab407ba988f99e5026d3da27524c/rdy2cpl/grids/base/orca.py#L76-L85
However, the plot above shows that there are some regions that require further care to compute the coordinates of the corners. This is not just the first row of points.  All the points in the rightmost column of black points oriented vertically would get wrong coordinates with the current method.
It is quite possible that all of this points are masked as land locations, but I do not know if having these inconsistent geometries may cause some problems to OASIS when using some conservative regrid algorithm.

I picked a slice extending from indices 30 to 50 in the first coordinate, and from 0 to 40 in the second.


For the ORCA1 grid, I get this other plot, which is consistent with the same interpretation:
![image](https://github.com/uwefladrich/rdy2cpl/assets/5628228/1d3e28f6-1f53-42bb-a072-395315105724)

**Comment by uwefladrich** [comment]
> [...] so I guess that `rdy2cpl` is currently only capable to adequately handle one of them (I guess eORCA1), even if both grids are listed in the `_orca_names` variable.

You are right, there is no (longer) support for the ORCA1 grid in rdy2cpl. It happened so that our ORCA1 --> eORCA1 switch coincided with our NEMO4.0 --> NEMO4.2 switch, which explains the missing doubled columns, see the [migration guide](https://sites.nemo-ocean.io/user-guide/migration.html).

The ORCA1 grid should be removed from rdy2cpl.
